core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddLocalExchangesForTaskScaleWriters.java

```diff
@@ -0,0 +1,110 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestAddLocalExchangesForTaskScaleWriters
+        extends BasePlanTest
+{
+    @Override
+    protected LocalQueryRunner createLocalQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("tiny")
+                .build();
+        LocalQueryRunner queryRunner = LocalQueryRunner.create(session);
+        queryRunner.createCatalog("tpch", new TpchConnectorFactory(1), ImmutableMap.of());
+        queryRunner.createCatalog("mock_dont_report_written_bytes", createConnectorFactorySupportingReportingBytesWritten(false, "mock_dont_report_written_bytes"), ImmutableMap.of());
+        queryRunner.createCatalog("mock_report_written_bytes", createConnectorFactorySupportingReportingBytesWritten(true, "mock_report_written_bytes"), ImmutableMap.of());
+        return queryRunner;
+    }
+
+    private MockConnectorFactory createConnectorFactorySupportingReportingBytesWritten(boolean supportsWrittenBytes, String name)
+    {
+        return MockConnectorFactory.builder()
+                .withSupportsReportingWrittenBytes(supportsWrittenBytes)
+                .withGetTableHandle(((session, schemaTableName) -> null))
+                .withName(name)
+                .build();
+    }
+
+    @DataProvider
+    public Object[][] taskScaleWritersEnabled()
+    {
+        return new Object[][] {{true}, {false}};
+    }
+
+    @Test(dataProvider = "taskScaleWritersEnabled")
+    public void testLocalScaledWriterDistributionWithSupportsReportingWrittenBytes(boolean isTaskScaleWritersEnabled)
+    {
+        Session session = testSessionBuilder()
+                .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, Boolean.toString(isTaskScaleWritersEnabled))
+                .setSystemProperty(SCALE_WRITERS, "false")
+                .build();
+
+        @Language("SQL")
+        String query = "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation";
+        SubPlan subPlan = subplan(query, LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED, false, session);
+        subPlan.getAllFragments().get(1).getRoot().getSources().forEach(node -> {
+            assertThat(node).isInstanceOf(ExchangeNode.class);
+            ExchangeNode exchangeNode = (ExchangeNode) node;
+            if (isTaskScaleWritersEnabled) {
+                assertThat(exchangeNode.getScope()).isEqualTo(LOCAL);
+                assertThat(exchangeNode.getPartitioningScheme().getPartitioning().getHandle()).isEqualTo(SCALED_WRITER_DISTRIBUTION);
+            }
+            else {
+                assertThat(exchangeNode.getPartitioningScheme().getPartitioning().getHandle()).isNotEqualTo(SCALED_WRITER_DISTRIBUTION);
+            }
+        });
+    }
+
+    @Test(dataProvider = "taskScaleWritersEnabled")
+    public void testLocalScaledWriterDistributionWithoutSupportsReportingWrittenBytes(boolean isTaskScaleWritersEnabled)
```

Comment:
You can just have one explicit test case with `isTaskScaleWritersEnabled=false`. No need to add `dataProvider` for this.